### PR TITLE
Modules

### PR DESCRIPTION
**Pull request by proddy**
Merged with feat_modules
To test you need to add -DEMSESP_TEST
Update AsyncTCP, ESPWebServer, MQTT libraries
Use Arduino 3.0.1
Also updated other small things in the code when I found them

### CODE REVIEW

**Comment by proddy** [comment]
@MichaelDvP if you find time could you test it?

**Review by MichaelDvP — APPROVED** [review]
A lot of changes, i have not read all the code. 
For espWebServer the  formatting has changed, hard to see the changes. 
I've just compiled and loaded in a s3-esp, works.

I still don't see the usecase for the modules, The examples can be enabled, are working and send a mqtt message. 
What i understand: They have a init on startup, a loop() in the main loop and can access emsesp-functions.

**Comment by proddy** [comment]
> A lot of changes, i have not read all the code. For espWebServer the formatting has changed, hard to see the changes. I've just compiled and loaded in a s3-esp, works.
> 
> I still don't see the usecase for the modules, The examples can be enabled, are working and send a mqtt message. What i understand: They have a init on startup, a loop() in the main loop and can access emsesp-functions.

yes - I'll send you a Discord PR on the use case, which is part of the other GitHub repo

File: src/system.h [hunk 1]
```diff
@@ -82,8 +82,6 @@ class System {
     void send_heartbeat();
     void send_info_mqtt();
 
-    std::string getMacAddress() const;
```
Comment (MichaelDvP) [inline src/system.h]: This causes link error,because it is refered by 
lib_deps = https://github.com/emsesp/EMS-ESP-Modules.git

Comment (proddy) [reply src/system.h]: i was working on the code, but didn't update the repo. Just did now, so should work.

I'm do a workaround for the breaking CONFIG_IDF_TARGET_* and then merge

**Comment by proddy** [comment]
@MichaelDvP with the latest dev.13 I just pushed there's a bug in MQTT Discovery which I'll need to fix... I've removed the firmware binaries from the GitHub releases

**Pull request merged**